src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -559,51 +559,31 @@ else if (value instanceof ByteBuffer) {
   public boolean shouldSkipNullValue(
       SinkRecord record,
       final SnowflakeSinkConnectorConfig.BehaviorOnNullValues behaviorOnNullValues) {
+    // do not skip when config is DEFAULT
     if (behaviorOnNullValues == SnowflakeSinkConnectorConfig.BehaviorOnNullValues.DEFAULT) {
       return false;
-    } else {
```

Comment:
the current logic in this method is as follows:
- do not skip record if the config allows null records
- skip record if the record value schema is a Snowflake converter schema AND the record value has isRecordContentValueNull 
- otherwise skip record if record value is null

The new logic in the method is similar, however we do the record value null check first